### ISSUE TITLE
'File is not included in module tree, ..' even after adding it to module tree.

### ISSUE THREAD

**Issue by TobiasDeBruijn** [intellij-rust/intellij-rust#9501]
## Environment

* **IntelliJ Rust plugin version:** 0.4.181.4928-222-nightly
* **Rust toolchain version:** 1.65.0-nightly (228710758 2022-09-10) x86_64-unknown-linux-gnu
* **IDE name and version:** CLion 2022.2.3 (CL-222.4167.35)
* **Operating system:** Linux 5.15.0-48-generic
* **Macro expansion:** enabled
* **Additional experimental features:** org.rust.cargo.emulate.terminal, org.rust.cargo.evaluate.build.scripts, org.rust.macros.proc

## Problem description
After creating a new file, the IDE reports that the file is not included in the module tree. This is correct behavior. After then adding the file to the parent `mod.rs`, the IDE does not pick this up and instead continuous telling the same message. Sometimes it's fixed by saving the new file, closing it, and reopening it. Other times it requires clearing the caches. It does not matter whether Cargo check for on-the-fly analyses is enabled.

I work in a multi-project Cargo workspace. I don't know if that is related to the issue.

## Steps to reproduce
Assume the following directory structure:
```
src/foo/mod.rs
src/foo/bar.rs
src/main.rs
```
`main.rs`
```rs
mod foo;
```

With `mod.rs` initially empty, the issue is reported, which is correct.
After changing `mod.rs` to
```rs
mod bar;
```
The 'File is not included in module tree' persists

**Comment by dima74** [comment]
Duplicate of #9405
